lib/arca/callback_analysis.rb

```diff
@@ -0,0 +1,128 @@
+module Arca
+  class CallbackAnalysis
+
+    # Arca::CallbackAnalysis takes an Arca::Model and data for a specific
+    # callback and then calculates and exposes a complete analysis for the
+    # callback including target methods, file paths, line numbers, booleans
+    # representing whether targets are in the same file they are called from,
+    # and finally the number of lines between callers and the target methods
+    # they call.
+    #
+    # model         - Arca::Model instance.
+    # callback_data - Hash with callback data collected by Arca::Collector.
+    def initialize(model:, callback_data:)
+      @model                          = model
+      @callback_symbol                = callback_data[:callback_symbol]
```

Comment:
I should have explained. If they are required then you should probably use callback_data.fetch(:callback_symbol). That is what I was going for.